instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/undertow/UndertowRequestContext.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.undertow;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/** Helper container for keeping track of request processing state in undertow. */
+public class UndertowRequestContext {
```

Comment:
Can you move it to `javaagent-api`? This class does not need to be available in the library instrumentations.

Also, nit: please make it final and add a private constructor.